three-nodes-hybrid-deployment.md

```diff
@@ -0,0 +1,130 @@
+
+In this example, the TPC-C 5000 Warehouse data is used in TiUP bench and the test lasts 12 hours with the `terminals` parameter set to `128` (concurrency). Close attention is paid to metrics related to performance stability of the cluster.
+
+The image below shows the QPS monitor of the cluster within 12 hours with the default parameter configuration. From the image, you can see an obvious performance jitter.
+
+![QPS with default config](/media/best-practices/three-nodes-default-config-qps.png)
+
+After the parameter adjustment, the performance is improved.
+
+![QPS with modified config](/media/best-practices/three-nodes-final-config-qps.png)
+
+## Parameter adjustment
+
+Performance jitter occurs in the image above, because the default thread pool configuration and the resource allocation to background tasks are for machines with sufficient resources. In the hybrid deployment scenario, the resources are shared among multiple components, so you need to limit the resource consumption via configuration parameters.
+
+The final cluster configuration for this test is as follows:
+
+{{< copyable "" >}}
+
+```yaml
+tikv:
+    readpool.unified.max-thread-count: 6
+    server.grpc-concurrency: 2
+    storage.scheduler-worker-pool-size: 2
+    gc.max-write-bytes-per-sec: 300K
+    rocksdb.max-background-jobs: 3
+    rocksdb.max-sub-compactions: 1
+    rocksdb.rate-bytes-per-sec: “200M”
+
+  tidb:
+    performance.committer-concurrency: 4
+    performance.max-procs: 8
+```
+
+The following sections introduce the meanings and the adjustment methods of these parameters.
+
+### Configuration of TiKV thread pool size
+
+This section offers best practices for adjusting parameters that relate to the resource allocation of thread pools for foreground applications. Reducing these thread pool sizes will compromise performance, but in the hybrid deployment scenario with limited resources, the cluster itself is hard to achieve high performance. In this scenario, the overall stability of the cluster is preferred over performance.
+
+If you conduct an actual load test, you can first use the default configuration and observe the actual resource usage of each thread pool. Then you can adjust the corresponding configuration items and reduce the sizes of the thread pools that have lower usage.
+
+#### `readpool.unified.max-thread-count`
+
+The default value of this parameter is 80% of the number of machine threads. In a hybrid deployment scenario, you need to manually calculate and specify this value. You can first set it to 80% of the expected number of CPU threads used by TiKV.
+
+#### `server.grpc-concurrency`
+
+This parameter defaults to `4`. Because in the existing deployment plan the CPU resources are limited, the actual requests are few. You can observe the monitoring panel, lower the value of this parameter, and keep the usage rate below 80%.
```

Comment:
```suggestion
This parameter defaults to `4`. Because in the existing deployment plan, the CPU resources are limited and the actual requests are few. You can observe the monitoring panel, lower the value of this parameter, and keep the usage rate below 80%.
```